lxd/db/cluster.go

```diff
@@ -0,0 +1,470 @@
+
+	n, err := result.RowsAffected()
+	if err != nil {
+		return fmt.Errorf("Fetch affected rows: %w", err)
+	}
+	if n != 1 {
+		return fmt.Errorf("Query deleted %d rows instead of 1", n)
+	}
+
+	return nil
+}
+
+// UpdateClusterGroup updates the ClusterGroup matching the given key parameters.
+// generator: ClusterGroup Update
+func (c *ClusterTx) UpdateClusterGroup(name string, object ClusterGroup) error {
+	id, err := c.GetClusterGroupID(name)
+	if err != nil {
+		return fmt.Errorf("Get ClusterGroup: %w", err)
+	}
+
+	stmt := c.stmt(clusterGroupUpdate)
+	result, err := stmt.Exec(object.Name, object.Description, id)
+	if err != nil {
+		return fmt.Errorf("Update ClusterGroup: %w", err)
+	}
+
+	n, err := result.RowsAffected()
+	if err != nil {
+		return fmt.Errorf("Fetch affected rows: %w", err)
+	}
+	if n != 1 {
+		return fmt.Errorf("Query updated %d rows instead of 1", n)
+	}
+
+	// Delete current nodes.
+	stmt = c.stmt(clusterGroupDeleteNodesRef)
+	_, err = stmt.Exec(id)
+	if err != nil {
+		return fmt.Errorf("Delete current nodes: %w", err)
+	}
+
+	// Insert nodes reference.
+	err = addNodesToClusterGroup(c.tx, int(id), object.Nodes)
+	if err != nil {
+		return fmt.Errorf("Insert nodes for cluster group: %w", err)
+	}
+
+	return nil
+}
+
+// ClusterGroupToAPI is a convenience to convert a ClusterGroup db struct into
+// an API cluster group struct.
+func ClusterGroupToAPI(clusterGroup *ClusterGroup, nodes []string) *api.ClusterGroup {
+	c := &api.ClusterGroup{
+		ClusterGroupPut: api.ClusterGroupPut{
+			Description: clusterGroup.Description,
+			Members:     nodes,
+		},
+		ClusterGroupPost: api.ClusterGroupPost{
+			Name: clusterGroup.Name,
+		},
+	}
+
+	return c
+}
+
+// GetClusterGroupNodes returns a list of nodes of the given cluster group.
+func (c *ClusterTx) GetClusterGroupNodes(groupName string) ([]string, error) {
+	q := `SELECT nodes.name FROM nodes_cluster_groups
+JOIN nodes ON nodes.id = nodes_cluster_groups.node_id
+JOIN cluster_groups ON cluster_groups.id = nodes_cluster_groups.group_id
+WHERE cluster_groups.name = ?`
+
+	return query.SelectStrings(c.tx, q, groupName)
+}
+
+// GetClusterGroupURIs returns all available ClusterGroup URIs.
+// generator: ClusterGroup URIs
+func (c *ClusterTx) GetClusterGroupURIs(filter ClusterGroupFilter) ([]string, error) {
+	var args []interface{}
+	var stmt *sql.Stmt
+	if filter.Name != nil && filter.ID == nil {
+		stmt = c.stmt(clusterGroupNamesByName)
+		args = []interface{}{
+			filter.Name,
+		}
+	} else if filter.ID == nil && filter.Name == nil {
+		stmt = c.stmt(clusterGroupNames)
+		args = []interface{}{}
+	} else {
+		return nil, fmt.Errorf("No statement exists for the given Filter")
+	}
+
+	code := cluster.EntityTypes["cluster group"]
+	formatter := cluster.EntityFormatURIs[code]
+
+	return query.SelectURIs(stmt, formatter, args...)
+}
+
+//AddNodeToClusterGroup adds a given node to the given cluster group.
+func (c *ClusterTx) AddNodeToClusterGroup(groupName, nodeName string) error {
```

Comment:
Please can you avoid the use of argument compression here, i.e it should be `groupName string, nodeName string` instead. Thanks